core/src/vaults.ts

```diff
@@ -0,0 +1,335 @@
+    const { currentCollateralDebtDai, maximumCollateralDebtDai, liquidationPenaltyRatio, minimalAuctionedDai } =
+        await fetchCollateralLiquidationLimitsAndLiquidatorAddress(network, vaultBase.collateralType);
+    return {
+        ...vaultBase,
+        ...vaultCollateralParameters,
+        ...vaultAmount,
+        ...globalLiquidationLimits,
+        currentCollateralDebtDai,
+        maximumCollateralDebtDai,
+        liquidationPenaltyRatio,
+        minimalAuctionedDai,
+    };
+};
+
+const _fetchLiquidationRatioAndOracleAddress = async (network: string, collateralType: CollateralType) => {
+    const contract = await getContract(network, 'MCD_SPOT');
+    const collateralTypeAsHex = ethers.utils.formatBytes32String(collateralType);
+    const oracleAndLiquidationRatio = await contract.ilks(collateralTypeAsHex);
+    const liquidationRatioAsHex = oracleAndLiquidationRatio.mat;
+    const oracleAddress = oracleAndLiquidationRatio.pip;
+    const liquidationRatio = new BigNumber(liquidationRatioAsHex._hex).shiftedBy(-RAY_NUMBER_OF_DIGITS);
+    return { oracleAddress, liquidationRatio };
+};
+
+export const fetchLiquidationRatioAndOracleAddress = memoizee(_fetchLiquidationRatioAndOracleAddress, {
+    maxAge: CACHE_EXPIRY_MS,
+    promise: true,
+    length: 2,
+});
+
+const _fetchVaultLiquidationIncentive = async (
+    network: string,
+    liquidatiorContractAddress: string,
+    debtDai: BigNumber
+) => {
+    const liquidatorContractInterface = await getContractInterfaceByName('MCD_CLIP_');
+    const provider = await getProvider(network);
+    const liquidatorContract = new ethers.Contract(liquidatiorContractAddress, liquidatorContractInterface, provider);
+    const incentiveConstantDaiHex = await liquidatorContract.tip();
+    const incentiveRelativeDaiHex = await liquidatorContract.chip();
+    const incentiveRelativeDai = new BigNumber(incentiveRelativeDaiHex._hex)
+        .shiftedBy(-WAD_NUMBER_OF_DIGITS)
+        .multipliedBy(debtDai);
+    const incentiveConstantDai = new BigNumber(incentiveConstantDaiHex._hex).shiftedBy(-RAD_NUMBER_OF_DIGITS);
+    const incentiveCombinedDai = incentiveRelativeDai.plus(incentiveConstantDai);
+    return {
+        incentiveCombinedDai,
+        incentiveConstantDai,
+        incentiveRelativeDai,
+    };
+};
+
+export const fetchVaultLiquidationIncentive = memoizee(_fetchVaultLiquidationIncentive, {
+    maxAge: CACHE_EXPIRY_MS,
+    promise: true,
+    length: 3,
+});
+
+const _fetchLiquidatedParameters = async (network: string, vault: Vault) => {
+    const contract = await getContract(network, 'MCD_DOG');
+    const typeHex = ethers.utils.formatBytes32String(vault.collateralType);
+    if (!vault.initialDebtDai.isZero()) {
+        return;
+    }
+    const eventFilter = contract.filters.Bark(typeHex, vault.address, null, null, null, null, null);
+    const liquidationEvents = await contract.queryFilter(eventFilter);
+    if (liquidationEvents.length !== 0) {
```

Comment:
Please prefer returning earlier, [as suggested before](https://github.com/sidestream-tech/unified-auctions-ui/pull/457#discussion_r977347070)
```suggestion
    if (liquidationEvents.length === 0) {
        return;
    }
```